src/services/profile/profile-router.test.ts

```diff
@@ -0,0 +1,231 @@
+import { describe, expect, it, beforeEach } from "@jest/globals";
+import Models from "../../database/models.js";
+import { TESTER, delAsUser, getAsAdmin, getAsUser, postAsAttendee, postAsUser } from "../../testTools.js";
+import { ProfileFormat } from "./profile-formats.js";
+import Constants from "../../constants.js";
+
+beforeEach(async () => {
+    Models.initialize();
+});
+
+const profile: ProfileFormat = {
+    userId: TESTER.id,
+    displayName: TESTER.name,
+    avatarUrl: TESTER.avatarUrl,
+    discordTag: TESTER.discordTag,
+    points: 0,
+};
+
+describe("POST /profile", () => {
+    it("posts for an attendee", async () => {
+        const response = await postAsAttendee("/profile/").send(profile).expect(200);
+
+        expect(JSON.parse(response.text)).toHaveProperty("displayName", TESTER.name);
+    });
+
+    it("posts for a user", async () => {
+        const response = await postAsUser("/profile/").send(profile).expect(200);
+
+        expect(JSON.parse(response.text)).toHaveProperty("displayName", TESTER.name);
+
+        // to verify they can't double create
+        const response2 = await postAsUser("/profile/").send(profile).expect(400);
+        expect(JSON.parse(response2.text)).toHaveProperty("error", "UserAlreadyExists");
+    });
+
+    it("posts with invalid data", async () => {
```

Comment:
np: `it("posts with invalid data")` -> `it("fails when invalid data is provided")`